zephyr/module.yml

```diff
@@ -1,2 +1,7 @@
 build:
   cmake: .
+  depends:
+    # Make sure that the nrf_hw_models module is processed before this one,
+    # so that its include paths are used first and hence certain nrfx files
+    # can be overriden when a simulated target is built.
+    - nrf_hw_models
```

Comment:
This doesn't seem to be a real dependency to me.
Users should be able to use `hal_nordic` without `nrf_hw_models`, correct ?

`nrf_hw_models` are only needed by babblesim users, if i'm not mistake.
Everyone else could decide not to include that module, but if they do this, trhey will now see this error:
```
  Unmet or cyclic dependencies in modules:

  /projects/github/ncs/modules/hal/nordic depends on: ['nrf_hw_models']
```

Include order should be handled in CMakeLists.txt.

Reply:
> This doesn't seem to be a real dependency to me.
> Users should be able to use hal_nordic without nrf_hw_models, correct ?

Yes. I think I misunderstood the way `depends:` is supposed to work when the module is not included.

> Include order should be handled in CMakeLists.txt.

How can I do it? I need the include paths from the `nrf_hw_models` module to appear in the compiler invocation before those from the `hal_nordic` module.

Reply:
I removed that dependency and applied [this change](https://github.com/BabbleSim/ext_NRF52_hw_models/pull/12/files#diff-af3b638bc2a3e6c650974192a53c7291) in the `nrf_hw_models` module instead. @tejlmand Could you take a look?

Reply:
> 
> Yes. I think I misunderstood the way `depends:` is supposed to work when the module is not included.
> 

Currently it does two things.
Ensure the module on which current module depends does exists, and it guarantees the order.

> > Include order should be handled in CMakeLists.txt.
> 
> How can I do it? I need the include paths from the `nrf_hw_models` module to appear in the compiler invocation before those from the `hal_nordic` module.

In CMake , it is possible to request an include path is appended BEFORE anything else:
https://cmake.org/cmake/help/latest/command/target_include_directories.html

Unfortunately this possibility is hidden in [zephyr_include_directories](https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/extensions.cmake#L76)
But this can still be achieved using CMake here:
https://github.com/BabbleSim/ext_NRF52_hw_models/blob/4d75a2a95197abb704fc0be25ec37e4992c4244d/CMakeLists.txt#L29-L32

with this code:
```
target_include_directories(zephyr_interface BEFORE  INTERFACE 
   ${CMAKE_CURRENT_LIST_DIR}/src/nrfx/hal/
   ${CMAKE_CURRENT_LIST_DIR}/src/HW_models/
)
```
should help you achieve want you want.

Reply:
> I removed that dependency and applied [this change](https://github.com/BabbleSim/ext_NRF52_hw_models/pull/12/files#diff-af3b638bc2a3e6c650974192a53c7291) in the `nrf_hw_models` module instead. @tejlmand Could you take a look?

Seems my browser window was outdated when replying with previous message.